test/extended/operators/olm.go

```diff
@@ -172,14 +173,30 @@ var _ = g.Describe("[sig-operator] an end user can use OLM", func() {
 		err = oc.AsAdmin().WithoutNamespace().Run("create").Args("-f", configFile).Execute()
 		o.Expect(err).NotTo(o.HaveOccurred())
 
-		o.Eventually(func() string {
-			output, err := oc.AsAdmin().Run("get").Args("-n", oc.Namespace(), "operatorgroup", "test-operator", "-o=jsonpath={.status.namespaces}").Output()
+		o.Eventually(func() []string {
+			// Using json output instead of jsonpath - oc/jsonpath bug seems to improperly decode `[""]` as `[]`
+			output, err := oc.AsAdmin().Run("get").Args("-n", oc.Namespace(), "operatorgroup", "test-operator", "-o=json").Output()
 			if err != nil {
 				e2e.Logf("Failed to get valid operatorgroup, error:%v", err)
-				return ""
+				return []string{""}
 			}
-			return output
-		}, 5*time.Minute, time.Second).Should(o.Equal("[]"))
+			type ogStatus struct {
+				Namespaces []string `json:"namespaces"`
+			}
+			type og struct {
+				Status ogStatus `json:"status"`
+			}
+			parsed := og{
+				Status: ogStatus{
+					Namespaces: []string{},
+				},
+			}
+			if err := json.Unmarshal([]byte(output), &parsed); err != nil {
+				e2e.Logf("Failed to parse operatorgroup, error:%v", err)
+				return []string{""}
+			}
+			return parsed.Status.Namespaces
+		}, 5*time.Minute, time.Second).Should(o.Equal([]string{""}))
```

Comment:
The solution from https://github.com/openshift/origin/pull/25351/files seems like a much better alternative.

Reply:
That solution will hide bugs where the actual api response is incorrect.